lib/hmr/lazyCompilationBackend.js

```diff
@@ -52,7 +60,16 @@ module.exports = (compiler, client, callback) => {
 		}
 		if (moduleActivated && compiler.watching) compiler.watching.invalidate();
 	};
-	const server = http.createServer(requestListener);
+
+	let server;
+	if (options && options.https) {
+		const https = require("https");
+		server = https.createServer(options.https, requestListener);
+	} else {
+		const http = require("http");
+		server = http.createServer(requestListener);
+	}
```

Comment:
There are more options for `http.createServer`, we should allow to pass of them, for `http` and `https`